File: Cracking Coding Interview/Through Python/LearningToSolving/src/Course/Module-1/Assignment/Task_1.py
class Task_1:
	def __init__(self, inpt):
		self.inpt = int(inpt)

	def solution(self) -> int:
		res = 1

		while self.inpt > 1:
			res *= self.inpt
			self.inpt -= 1

		return res


inpt = input('Enter number: ')
o1 = Task_1(inpt)
print('The factorial result: ', o1.solution())


# Time Complexity: O(n)
# Space Complexity: O(1)

File: Cracking Coding Interview/Through Java/LearningToSolving/src/Course/Module_1/Assignment/Task_1.java
package Course.Module_1.Assignment;

import java.util.Scanner;

public class Task_1 {
    // Find the factorial of a given number
    void solution() {
        Scanner sc = new Scanner(System.in);

        System.out.print("Enter input number: ");
        int input_number = sc.nextInt();

        int result = 1;
        while (input_number > 1) {
            result *= input_number;
            input_number--;
        }

        System.out.println("The result of factorial is " + result);
    }

    // Main method
    public static void main(String[] args) {
        Task_1 task1 = new Task_1();
        task1.solution();
    }
}


// Time Complexity : O(n)
// Space Complexity : O(1)
